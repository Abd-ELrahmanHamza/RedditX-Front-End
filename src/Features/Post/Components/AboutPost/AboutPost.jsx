import profile from "Features/Post/Assets/Images/profile.jpg";
import { HiOutlineDocumentText } from "react-icons/hi";
import { Link, useNavigate } from "react-router-dom";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Tooltip from "react-bootstrap/Tooltip";
import {
  AboutPostContainer,
  BackgroundImage,
  Date,
  Online,
  Paragraph,
  Profile,
  ProfileLink,
  Statistics,
  Button,
} from "./AboutPost.styled";
import { useState } from "react";
import CommunityOptions from "Features/Post/Components/CommunityOptions/CommunityOptions";

const AboutPost = () => {
  const [toggleStatus, setToggleStatus] = useState("Joined");

  const onMouseOver = (e) => {
    e.preventDefault();
    setToggleStatus("Leave");
  };
  const onMouseLeave = (e) => {
    e.preventDefault();
    setToggleStatus("Joined");
  };
  const navigate = useNavigate();

  return (
    <AboutPostContainer>
      <div className="content">
        <BackgroundImage />
        <Profile>
          <img src={profile} alt={profile} />
          <ProfileLink>
            <Link to={"/"}>
              <span>r/Egypt</span>
            </Link>
          </ProfileLink>
        </Profile>
        <Paragraph>
          <div>
            Welcome to r/Egypt. The Nile River, the birthplace of civilization,
            and the home of the ancient Pharaohs. We're a country-themed
            subreddit dedicated to posting on events that affect Egypt and
            Egyptians. We're a multilingual subreddit, therefore submissions
            made in either English or Arabic are welcome.
          </div>
        </Paragraph>
        <Date>
          <i>
            <HiOutlineDocumentText size={25} />
          </i>
          <OverlayTrigger
            placement={"bottom"}
            overlay={<Tooltip>14 years ago</Tooltip>}
          >
            <span>Aug 12, 2008</span>
          </OverlayTrigger>
        </Date>
        <hr />
        <Statistics>
          <OverlayTrigger
            placement={"bottom"}
            overlay={<Tooltip>Pharaohs</Tooltip>}
          >
            <div>
              <div className={"members"}>180k</div>
              <p>Pharaohs</p>
            </div>
          </OverlayTrigger>
          <OverlayTrigger
            placement={"bottom"}
            overlay={<Tooltip>285 Masry</Tooltip>}
          >
            <div>
              <Online>285</Online>
              <p>Masry</p>
            </div>
          </OverlayTrigger>
          <OverlayTrigger
            placement={"bottom"}
            overlay={
              <Tooltip>
                r/Egypt in the top of largest communities on Reddit{" "}
              </Tooltip>
            }
          >
            <Link to={"https://www.reddit.com/best/communities/4/#t5_2ql2m"}>
              <span>
                <div className={"top"}>Top 1%</div>
                <p>Ranked by size</p>
              </span>
            </Link>
          </OverlayTrigger>
        </Statistics>
        <hr />
        <div>
          <Button but={"1"} onMouseOver={onMouseOver} onMouseOut={onMouseLeave}>
            {toggleStatus}
          </Button>
          <Button onClick={() => navigate("/submit")}>Create Post</Button>
          <hr />
          <CommunityOptions />
        </div>
      </div>
    </AboutPostContainer>
  );
};

export default AboutPost;
