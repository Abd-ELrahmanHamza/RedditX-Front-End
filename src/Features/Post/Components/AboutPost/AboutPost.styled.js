import styled from "styled-components";
import egyptImg from "Features/Post/Assets/Images/Backgorund-post.png";

export const AboutPostContainer = styled.div`
  border: 1px solid ${({ theme }) => theme.background.primary};
  border-radius: 4px;
  overflow: visible;
  overflow-wrap: break-word;
  cursor: pointer;

  .content {
    max-height: none;
    padding: 12px;
    background-color: ${({ theme }) => theme.background.primary};
  }

  hr {
    background-color: ${({ theme }) => theme.color.gray};
    border: none;
    height: 1px;
    margin: 16px 0;
  }

`;

export const BackgroundImage = styled.div`
  background-color: #ea0027;
  background-size: cover;
  background-image: url(${egyptImg});
  background-position-y: center;
  background-position-x: center;
  background-repeat: no-repeat;
  border-radius: 3px 3px 0 0;
  height: 34px;
  margin: -12px -12px 10px;
`;

export const Profile = styled.div`
  align-items: center;
  display: flex;
  margin-bottom: 8px;

  img {
    border-radius: 100%;
    vertical-align: middle;
    background-position: 50%;
    background-repeat: no-repeat;
    background-size: 100%;
    height: 54px;
    width: 54px;
    font-size: 54px;
    line-height: 54px;
    box-sizing: border-box;
    flex: none;
    margin-right: 8px;
    background-color: rgb(0, 121, 211);
  }
`;

export const ProfileLink = styled.div`
  display: inline-block;
  vertical-align: middle;

  a {
    flex: 1 1 auto;
    overflow: hidden;
    text-overflow: ellipsis;
    color: inherit;
    text-decoration: inherit;
  }

  span {
    font-size: 16px;
    font-weight: 500;
    line-height: 20px;
    display: inline-block;
  }
`;

export const Paragraph = styled.div`
  margin-bottom: 8px;
  position: relative;

  :first-child {
    font-family: Noto Sans, Arial, sans-serif;
    font-size: 14px;
    line-height: 21px;
    font-weight: 400;
    word-wrap: break-word;
  }
`
export const Date = styled.div`
  margin-top: 12px;
  margin-bottom: 8px;

  :first-child {
    font-family: Noto Sans, Arial, sans-serif;
    font-size: 14px;
    font-weight: 400;
    line-height: 18px;
    display: flex;
    flex-flow: row nowrap;
  }

  i {
    margin: -2px 8px 0 0;
    font-size: 20px;
    font-weight: 400;
    height: 20px;
    line-height: 20px;
    width: 20px;
  }

  span {
    color: ${({ theme }) => theme.color.muted};
  }
`
export const Statistics = styled.div`
  display: flex;
  justify-content: space-between;

  p {
    font-weight: 400;
    color: #747576;
    font-size: 12px;
    line-height: 16px;
    display: inline-block;
    word-break: break-word;
  }

  .top {
    font-size: 16px;
    font-weight: 500;
    line-height: 20px;
    display: block;
    color: ${({ theme }) => theme.color.primary};
  }

  .members {
    font-size: 16px;
    font-weight: 500;
    line-height: 20px;
  }

  a {
    text-decoration: none;
  }
`
export const Online = styled.div`
  font-size: 16px;
  font-weight: 500;
  line-height: 20px;

  :before {
    content: "●";
    margin-right: 4px;
    color: #46d160
  }
`
export const Button = styled.button`
  position: relative;
  margin-bottom: 8px;
  background-color:${({ but }) => but === "1" ? "white" : "#0079d3"};
  color: ${({ but }) => but === "1" ? "#0079d3" : "white"};
  justify-content: center;
  margin-top: 12px;
  border: 1px solid #0079d3;
  border-radius: 9999px;
  height: 30px;
  width: 100%;
  font-family: Noto Sans, Arial, sans-serif;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  line-height: 17px;
  text-transform: unset;
  min-height: 32px;
  min-width: 32px;
  padding: 4px 16px;

  :hover {
    background-color: ${({ theme }) => theme.background.hoverLight};
  }
`