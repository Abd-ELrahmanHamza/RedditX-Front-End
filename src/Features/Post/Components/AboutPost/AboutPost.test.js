import { shallow } from "enzyme";
import React from "react";

// Import components
import AboutPost from "./AboutPost";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockedUsedNavigate,
}));

describe("About Post", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<AboutPost />);
    });

    it("should render without crashing", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
