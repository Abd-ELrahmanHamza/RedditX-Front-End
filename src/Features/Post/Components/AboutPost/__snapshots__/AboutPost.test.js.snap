// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Post should render without crashing 1`] = `
<styled.div>
  <div
    className="content"
  >
    <styled.div />
    <styled.div>
      <img
        alt="profile.jpg"
        src="profile.jpg"
      />
      <styled.div>
        <Link
          to="https://www.reddit.com/r/Egypt/"
        >
          <span>
            r/Egypt
          </span>
        </Link>
      </styled.div>
    </styled.div>
    <styled.div>
      <div>
        Welcome to r/Egypt. The Nile River, the birthplace of civilization, and the home of the ancient Pharaohs. We're a country-themed subreddit dedicated to posting on events that affect Egypt and Egyptians. We're a multilingual subreddit, therefore submissions made in either English or Arabic are welcome.
      </div>
    </styled.div>
    <styled.div>
      <i>
        <HiOutlineDocumentText
          size={25}
        />
      </i>
      <OverlayTrigger
        defaultShow={false}
        overlay={
          <Tooltip
            placement="right"
          >
            14 years ago
          </Tooltip>
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <span>
          Aug 12, 2008
        </span>
      </OverlayTrigger>
    </styled.div>
    <hr />
    <styled.div>
      <OverlayTrigger
        defaultShow={false}
        overlay={
          <Tooltip
            placement="right"
          >
            Pharaohs
          </Tooltip>
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <div>
          <div
            className="members"
          >
            180k
          </div>
          <p>
            Pharaohs
          </p>
        </div>
      </OverlayTrigger>
      <OverlayTrigger
        defaultShow={false}
        overlay={
          <Tooltip
            placement="right"
          >
            285 Masry
          </Tooltip>
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <div>
          <styled.div>
            285
          </styled.div>
          <p>
            Masry
          </p>
        </div>
      </OverlayTrigger>
      <OverlayTrigger
        defaultShow={false}
        overlay={
          <Tooltip
            placement="right"
          >
            r/Egypt in the top of largest communities on Reddit
             
          </Tooltip>
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <Link
          to="https://www.reddit.com/best/communities/4/#t5_2ql2m"
        >
          <span>
            <div
              className="top"
            >
              Top 1%
            </div>
            <p>
              Ranked by size
            </p>
          </span>
        </Link>
      </OverlayTrigger>
    </styled.div>
    <hr />
    <div>
      <styled.button
        but="1"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        Joined
      </styled.button>
      <styled.button
        onClick={[Function]}
      >
        Create Post
      </styled.button>
      <hr />
      <CommunityOptions />
    </div>
  </div>
</styled.div>
`;
